.github/workflows/test.yaml

```diff
@@ -0,0 +1,99 @@
+on:
+  workflow_call:
+    inputs:
+      artefact:
+        required: true
+        type: string
+      tag:
+        required: true
+        type: string
+
+jobs:
+  test:
+    name: Test with coverage
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, self-hosted]
+
+    steps:
+      # region container setup
```

Comment:
I generally don't love `# region`'s for any file that isn't enormous, it just feels like bloat to me. But keep them if you feel strongly about them.

Reply:
I will remove them. This was left over from earlier attempts when this section was quite a bit longer.

Reply:
Removed in e08183e